src/Interpreters/AsynchronousInsertQueue.cpp

```diff
@@ -485,6 +485,8 @@ void AsynchronousInsertQueue::finishWithException(
             entry->finish(std::make_exception_ptr(exception));
         }
     }
+
+    CurrentMetrics::sub(CurrentMetrics::PendingAsyncInsert, entries.size());
```

Comment:
Should not we subtract only entries size that was not finished ?

Reply:
No, we subtract size of all entries at once [here](https://github.com/ClickHouse/ClickHouse/blob/ea3ccd2431c810dd0d28234d5e71959aefae4ee8/src/Interpreters/AsynchronousInsertQueue.cpp#L454) if there wasn't an error and in this line if there was one.